docs/api/utils.rst

```diff
@@ -0,0 +1,604 @@
+
+   .. method:: GetTextLength(string text, string fontName: "helv", float fontSize: 11, int encoding: 0)
+
+      Calculate the length of text on output with a given **builtin** font, :data:`fontSize` and encoding.
+
+      :arg string text: the text string.
+      :arg string fontName: the fontName. Must be one of either the :ref:`Base-14-Fonts` or the CJK fonts, identified by their "reserved" fontnames (see table in :meth:`Page.InsertFont`).
+      :arg float fontSize: the :data:`fontSize`.
+      :arg int encoding: the encoding to use. Besides 0 = Latin, 1 = Greek and 2 = Cyrillic (Russian) are available. Relevant for Base-14 fonts "Helvetica", "Courier" and "Times" and their variants only. Make sure to use the same value as in the corresponding text insertion.
+      :rtype: float
+      :returns: the length in points the string will have (e.g. when used in :meth:`Page.InsertText`).
+
+      .. note:: This function will only do the calculation -- it won't insert font nor text.
+
+      .. note:: The :ref:`Font` class offers a similar method, :meth:`Font.text_length`, which supports Base-14 fonts and any font with a character map (CMap, Type 0 fonts).
+
+      .. warning:: If you use this function to determine the required rectangle width for the (:ref:`Page` or :ref:`Shape`) *InsertTextbox* methods, be aware that they calculate on a **by-character level**.
+
+-----
+
+   .. method:: GetPdfString(string text)
+
+      Make a PDF-compatible string: if the text contains code points *Convert.ToInt32(c) > 255*, then it will be converted to UTF-16BE with BOM as a hexadecimal character string enclosed in "<>" brackets like *<feff...>*. Otherwise, it will return the string enclosed in (round) brackets, replacing any characters outside the ASCII range with some special code. Also, every "(", ")" or backslash is escaped with a backslash.
+
+      :arg string text: the object to convert
+
+      :rtype: string
+      :returns: PDF-compatible string enclosed in either *()* or *<>*.
+
+-----
+
+   .. method:: GetImageProfile(byte[] stream)
+
+      Show important properties of an image provided as a memory area. Its main purpose is to avoid using other Python packages just to determine them.
+
+      :arg bytes stream: either an image in memory or an **opened** file. An image in memory may be any of the formats `bytes`.
+
+      :rtype: ImageInfo
+      :returns:
+         No exception is ever raised. In case of an error, `None` is returned. Otherwise, there are the following items::
+
+         There is the following relation to **Exif** information encoded in `orientation`, and correspondingly in the `transform` matrix-like (quoted from MuPDF documentation, *ccw* = counter-clockwise):
+
+            0. Undefined
+            1. 0 degree ccw rotation. (Exif = 1)
+            2. 90 degree ccw rotation. (Exif = 8)
+            3. 180 degree ccw rotation. (Exif = 3)
+            4. 270 degree ccw rotation. (Exif = 6)
+            5. flip on X. (Exif = 2)
```

Comment:
Why can't we capitilize the first letter of "flip"? earlier on 0. we did `0. Undefined` so why not `5. Flip on X` etc. ?